src/Psalm/Type/Algebra.php

```diff
@@ -651,9 +651,7 @@ public static function groupImpossibilities(array $clauses): array
 
                         $new_clause = new Clause(
                             $new_clause_possibilities,
-                            $clause->creating_conditional_id === $grouped_clause->creating_conditional_id
-                                ? $clause->creating_conditional_id
-                                : $grouped_clause->creating_conditional_id,
+                            $grouped_clause->creating_conditional_id,
```

Comment:
On the first glance this doesn't look like exact equivalent.

Reply:
Really? Here's the step by step:
1. `$clause->creating_conditional_id === $grouped_clause->creating_conditional_id ? $clause->creating_conditional_id : $grouped_clause->creating_conditional_id,`
2. If `$clause->creating_conditional_id === $grouped_clause->creating_conditional_id`, then the two can be swapped
3. `$clause->creating_conditional_id === $grouped_clause->creating_conditional_id ? $grouped_clause->creating_conditional_id : $grouped_clause->creating_conditional_id,`
4. Then, the two results are the same so the ternary is useless
5. `$grouped_clause->creating_conditional_id` is enough

Reply:
If it required five points to explain it's wasn't immediately obvious. Everybody hates that math teacher that says 'now it's trivially obvious that ...' and skips a page or two of the proof :laughing: